### DISCUSSION TITLE
Doubt Related to How IM Process the image

### DISCUSSION THREAD

**Discussion by programmerarunkumar** [ImageMagick/ImageMagick#5086] [Help]
IM will store the image in memory if there is enought space, otherwise they are stored in Disk

Is there any way to find that the image is stored in memory or Disk using MagickWand for logging purpose?

**Comment by snibgo** [comment]
I suggest either: 

(1) While the command is running, look in your temporary directory. If IM has created a file in that directory, that is (probably) the pixel cache.

(2) Include "-debug cache" in your command. Search for lines containing "disk" or "Disk" like this ...
```
2022-04-26T15:42:11+00:00 0:00.027 0.031u 7.1.0 Cache magick[6252]: cache.c/SetPixelCacheExtent/3599/Cache
  extend [0] (C:/Users/Alan/AppData/Local/Temp/magick-OLYSVVfHQPVeN01MRff4qmBchsVFwUPQ[3], disk, 12B)
2022-04-26T15:42:11+00:00 0:00.029 0.031u 7.1.0 Cache magick[6252]: cache.c/OpenPixelCache/3952/Cache
  open [0] (C:/Users/Alan/AppData/Local/Temp/magick-OLYSVVfHQPVeN01MRff4qmBchsVFwUPQ[-1], Map, 1x1x3 12B)
2022-04-26T15:42:11+00:00 0:00.034 0.031u 7.1.0 Cache magick[6252]: cache.c/SetPixelCacheExtent/3599/Cache
  extend [0] (C:/Users/Alan/AppData/Local/Temp/magick-7Ql3DXBYQI6vqy-gWek91rhqHy-xzPtC[3], disk, 12B)
2022-04-26T15:42:11+00:00 0:00.042 0.047u 7.1.0 Cache magick[6252]: cache.c/SetPixelCacheExtent/3599/Cache
  extend [0] (C:/Users/Alan/AppData/Local/Temp/magick-lUHUKNPJroxqM-8pk5XS5Fo4ZMhL6GdK[3], disk, 12B)
2022-04-26T15:42:11+00:00 0:00.043 0.047u 7.1.0 Cache magick[6252]: cache.c/OpenPixelCache/3981/Cache
  open [0] (C:/Users/Alan/AppData/Local/Temp/magick-lUHUKNPJroxqM-8pk5XS5Fo4ZMhL6GdK[3], Disk, 1x1x3 12B)
```

**Reply by programmerarunkumar** [reply]
Is there way to get the info in Program using MagickWand?

**Reply by programmerarunkumar** [reply]
> 
I have tried -debug cache in command line for resizing the image 
Here how to find where IM stored the image data from this [Info](https://drive.google.com/file/d/1aA4P4FmM8RqK7sh6DnK6jVy1bqUMmtQQ/view?usp=sharing)

**Comment by snibgo** [comment]
> Is there way to get the info in Program using MagickWand?

IM is open-source. IM decides whether to store an image cache in memory or on disk. A program written for IM could access that information. I haven't done that, personally.

> Here how to find where IM stored the image data from this [LINK]Info

None of those text lines contain "disk" or "Disk", so the cache is stored in memory.

**Comment by programmerarunkumar** [comment]
Let me know if someone know how to find that the image is stored in memory or Disk using MagickWand in program 

FYI : Im expecting the solution for this in program not in command Line.

**Comment by urban-warrior** [comment]
MagickWand does not return the storage disposition of the image, however, you can call MagickCore methods from a MagickWand program. Call GetImagePixelCacheType() to return the pixel cache type: Memory, Disk, etc.
